src/api/app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,92 @@
+class ReportsController < ApplicationController
+  validate_action show: { response: :report }
+  validate_action create: { method: :post, request: :report }
+  validate_action update: { method: :put, request: :report }
+
+  before_action :set_report, only: %i[show update destroy]
+
+  # GET /reports
+  def index
+    @reports = Report.order(:id)
+    authorize @reports
+
+    user = User.session
+    if user.present? && !(user.admin? || user.moderator? || user.staff?)
+      @reports = @reports.where(reporter: user)
+    end
```

Comment:
https://github.com/varvet/pundit?tab=readme-ov-file#scopes

Reply:
It is BTW also *not* possible that `User.session` isn't present, we require login for all of the API.

Reply:
True. :+1:

I added a commit changing the code to use a Pundit scope. Please take a look again.